src/main/java/com/exactpro/th2/lib/template/metric/PrometheusMetricUtils.java

```diff
@@ -0,0 +1,51 @@
+package com.exactpro.th2.lib.template.metric;
+
+import io.prometheus.client.Histogram;
+import kotlin.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.LongAdder;
+
+public class PrometheusMetricUtils {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PrometheusMetricUtils.class.getName());
+
+    private final ScheduledExecutorService scheduler;
+    private final Histogram histogram;
+    private final ConcurrentMap<String, Pair<Histogram.Child, LongAdder>> counterMap = new ConcurrentHashMap<>();
+
+    public PrometheusMetricUtils(ScheduledExecutorService scheduledExecutorService,
+                                 Histogram prometheusHistogram,
+                                 long intervalMs) {
+        this.scheduler = scheduledExecutorService;
+        this.histogram = prometheusHistogram;
+
+        scheduler.scheduleAtFixedRate(() -> {
```

Comment:
Save the value in an internal variable and call the ScheduledFuture.close on close instance of the current class